common/static/common/js/discussion/discussion_router.js

```diff
@@ -129,9 +128,7 @@
             DiscussionRouter.prototype.navigateToThread = function(thread_id) {
                 var thread;
                 thread = this.discussion.get(thread_id);
-                return this.navigate("" + (thread.get("commentable_id")) + "/threads/" + thread_id, {
-                    trigger: true
-                });
+                return this.showThread(thread);
```

Comment:
@itsjeyd But the new code is still using `this.discussion.get(thread_id);` - it's just doing that lookup in `navigateToThread` instead of in `showThread`. So I don't see why this would fix it?